databuilder/example/scripts/sample_snowflake_table_usage.scala

```diff
@@ -0,0 +1,131 @@
+"""
+This is a example script that parses table names from Snowflake's query logs and calculates their popularity based on
+how many times that same user queries that specific table.
+"""
+
+def getTables(query: String): Seq[String] = {
+  val logicalPlan = spark.sessionState.sqlParser.parsePlan(query)
+  import org.apache.spark.sql.catalyst.analysis.UnresolvedRelation
+  logicalPlan.collect { case r: UnresolvedRelation => r.tableName }
+}
+
+import java.sql.Connection
+import java.sql.DriverManager
+import java.sql.Statement
+import java.sql.ResultSet
+import java.util.Base64
+import java.util.Properties
+import java.security.KeyFactory
+import java.security.PrivateKey
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo
+import java.security.Security
+import java.security.spec.PKCS8EncodedKeySpec
+import scala.collection.mutable.Map
+import scala.collection.mutable.Set
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import org.apache.spark.sql.catalyst.parser.ParseException
+import org.bouncycastle.openssl.PEMParser
+import java.io.StringReader
+import org.bouncycastle.openssl.jcajce.JcaPEMWriter
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter
+import org.bouncycastle.openssl.jcajce.JcePEMDecryptorProviderBuilder
+import org.bouncycastle.openssl.jcajce.JceOpenSSLPKCS8DecryptorProviderBuilder
+
+Security.removeProvider("BC")
+Security.insertProviderAt(new BouncyCastleProvider(), 0)
+
+Class.forName("net.snowflake.client.jdbc.SnowflakeDriver")
+
+val decryptedPrivateKey = EncryptionUtils.decryptPrivateKey(pem, pwd)
+
+val pkcs8EncodedBytes = Base64.getDecoder().decode(decryptedPrivateKey)
+
+val keySpec = new PKCS8EncodedKeySpec(pkcs8EncodedBytes)
+val kf = KeyFactory.getInstance("RSA")
+val privateKey = kf.generatePrivate(keySpec)
+
+val url = "jdbc:snowflake://snowflake_account_name.snowflakecomputing.com";
+val prop = new Properties();
+
+// TODO: change these values to your snowflake creds
+val user = "user@email.com"
```

Comment:
interesting, to make sure I understand correctly: you run a databricks scala job to read and calculate snowflake table usage?

Reply:
yup! exactly. We also store our creds (user and private key) in databricks secrets 😄 it's super cool. Scala isn't my language of choice, but it had the best sql parser package.

Reply:
nice! we are doing very similar thing but with Apache Ranger audit logs, which makes it easier cause they are already json serialized.